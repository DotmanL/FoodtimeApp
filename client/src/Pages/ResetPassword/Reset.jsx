import React, { useState, useEffect, Fragment } from 'react';
import {
  Container,
  FormInput,
  Title,
  FormContainer,
  ButtonContainer,
} from './Reset.styles';
import { reset } from '../../Redux/User/user.actions';
import jwt from 'jsonwebtoken';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import CustomButton from '../../components/custom-button/custom-button';

export const Reset = ({ match, reset }) => {
  const [inputs, setInputs] = useState({
    firstName: '',
    newPassword: '',
    confirmPassword: '',
  });

  useEffect(() => {
    let token = match.params.token;
    let { firstName } = jwt.decode(token);
    if (token) {
      setInputs({ firstName });
    }
  }, [match.params.token]);

  const { newPassword, firstName, confirmPassword } = inputs;

  const handleChange = (event) => {
    const { name, value } = event.target;
    setInputs({ ...inputs, [name]: value });
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    setInputs({ newPassword: '', confirmPassword: '' });
    if (newPassword !== confirmPassword) {
      toast.error("Passwords don't match");
    } else {
      reset(match.params.token, newPassword);
    }
  };

  return (
    <Fragment>
      <Container>
        <FormContainer>
          <Title>Hey, {firstName} type in your new password </Title>

          <form onSubmit={handleSubmit}>
            <FormInput
              name="newPassword"
              type="password"
              value={newPassword}
              onChange={handleChange}
              placeholder="New Password"
              minLength="6"
              required
            />

            <FormInput
              name="confirmPassword"
              type="password"
              value={confirmPassword}
              onChange={handleChange}
              placeholder="Re-enter New Password"
              minLength="6"
              required
            />
            <ButtonContainer>
              <CustomButton type="submit"> Submit </CustomButton>
            </ButtonContainer>
          </form>
        </FormContainer>
      </Container>
    </Fragment>
  );
};

Reset.propType = {
  reset: PropTypes.func.isRequired,
};

export default connect(null, { reset })(Reset);
