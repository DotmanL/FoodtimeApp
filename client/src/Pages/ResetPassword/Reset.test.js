import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { Reset } from './Reset'

describe('Confirmation component', () => {
  let wrapper;
  let mockReset;
  let mockSubmit;

  beforeEach(() => {
    mockReset = jest.fn();
    mockSubmit = jest.fn();
    const match = { params: {
      id: 1243466,
    },
  }
    const mockProps = {
      reset: mockReset,
      handleSubmit: mockSubmit
    };

    wrapper = shallow(<Reset {...mockProps} match={match} />);
  });

  it('should render Confirmation Page component', () => {
    expect(toJson(wrapper))
    .toMatchSnapshot();
  })
  
  it('should simulate custom button click', () => {
    wrapper.find('CustomButton').simulate('click')
    expect(mockReset).toHaveBeenCalledTimes(0);
  
  })
  
  it('should simulate handlesubmit on button click', () => {
    wrapper.find('CustomButton').simulate('submit', { preventDefault () {} });
  
  })     

})