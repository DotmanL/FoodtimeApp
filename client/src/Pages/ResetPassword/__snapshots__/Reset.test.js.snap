// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation component should render Confirmation Page component 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.h2>
        Hey, 
         type in your new password 
      </styled.h2>
      <form
        onSubmit={[Function]}
      >
        <styled.input
          minLength="6"
          name="newPassword"
          onChange={[Function]}
          placeholder="New Password"
          required={true}
          type="password"
          value=""
        />
        <styled.input
          minLength="6"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Re-enter New Password"
          required={true}
          type="password"
          value=""
        />
        <CustomButton
          type="submit"
        >
           Submit 
        </CustomButton>
      </form>
    </styled.div>
  </styled.div>
</Fragment>
`;
